operator/src/main/resources/kafka-metrics.yaml

```diff
@@ -305,3 +305,64 @@ data:
         pattern: >-
           kafka.server<type=KafkaServer, name=BrokerState><>Value: (\d+)
         type: GAUGE
+      # OAuth Metrics - from https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/main/testsuite/docker/kafka/config/metrics-config.yml
+      - pattern: "strimzi.oauth<type=(.+), context=(.+), kind=(.+), host=\"(.+)\", path=\"(.+)\", (.+)=(.+), (.+)=(.+), (.+)=(.+)><>(count|totalTimeMs):"
+        name: "strimzi_oauth_$1_$12"
+        type: COUNTER
+        labels:
+          context: "$2"
+          kind: "$3"
+          host: "$4"
+          path: "$5"
+          "$6": "$7"
+          "$8": "$9"
+          "$10": "$11"
+      - pattern: "strimzi.oauth<type=(.+), context=(.+), kind=(.+), host=\"(.+)\", path=\"(.+)\", (.+)=(.+), (.+)=(.+)><>(count|totalTimeMs):"
+        name: "strimzi_oauth_$1_$10"
+        type: COUNTER
+        labels:
+          context: "$2"
+          kind: "$3"
+          host: "$4"
+          path: "$5"
+          "$6": "$7"
+          "$8": "$9"
+      - pattern: "strimzi.oauth<type=(.+), context=(.+), kind=(.+), host=\"(.+)\", path=\"(.+)\", (.+)=(.+)><>(count|totalTimeMs):"
+        name: "strimzi_oauth_$1_$8"
+        type: COUNTER
+        labels:
+          context: "$2"
+          kind: "$3"
+          host: "$4"
+          path: "$5"
+          "$6": "$7"
```

Comment:
I wonder if these three entries are redundant? The 3 patterns below look like more generalised forms that would also cover these cases?

Reply:
No, they won't be redundant.  They are matching different forms. The number of groups is different in each case.

Reply:
I had counted the number of groups in each case, and they appeared to be the same is what I meant. :slightly_smiling_face:  It seems like the actual difference is that the above 3 entries are counters, while the below 3 are gauges. Here's a diff tool view with the above on the left, and the below on the right:

![image](https://user-images.githubusercontent.com/442386/233071970-dce30e45-bc37-440c-a2fc-ccdc5c18a545.png)